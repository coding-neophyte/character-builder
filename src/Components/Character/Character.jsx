import React from 'react'

export default function Character({ head, torso, legs }) {
    return (
        <div style={{display: 'flex', justifyContent: 'center' }}>
        <section style={{display: 'flex', flexDirection: 'column'}}>
            <article aria-label='headArticle'className='head' style={{
                backgroundImage: `url(./${head}.png)`,
                height: 150,
                width: 150,
                backgroundSize: 'cover',
                backgroundRepeat: 'no-repeat',}}> </article>


            <article aria-label='torsoArticle' className='torso'style={{
                backgroundImage: `url(./${torso}.png)`,
                height: 150,
                width: 150,
                backgroundSize: 'cover',
                backgroundRepeat: 'no-repeat'}}> </article>


            <article aria-label='legsArticle' className='legs'style={{
                backgroundImage: `url(./${legs}.png)`,
                height: 150,
                width: 150,
                backgroundSize: 'cover',
                backgroundRepeat: 'no-repeat',}}> </article>

        </section>
        </div>
    )
}
