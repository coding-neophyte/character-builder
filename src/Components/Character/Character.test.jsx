import { screen, render } from '@testing-library/react'
import Character from './Character'

it('should render elements in character component', () => {
    const {container} = render(<Character head='beanie' torso='cardigan' legs='baggy'  />)

    const selectHead = screen.getByLabelText('headArticle')
    const selectTorso = screen.getByLabelText('torsoArticle')
    const selectLegs = screen.getByLabelText('legsArticle')


    expect(selectHead).toHaveStyle(`backgroundImage: url(./beanie.png)`)
    expect(selectTorso).toHaveStyle(`backgroundImage: url(./cardigan.png)`)
    expect(selectLegs).toHaveStyle(`backgroundImage: url(./baggy.png)`)
    expect(container).toMatchSnapshot();
})
