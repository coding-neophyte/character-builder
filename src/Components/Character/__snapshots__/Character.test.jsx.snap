// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render elements in character component 1`] = `
<div>
  <section
    style="display: flex; flex-direction: column;"
  >
    <article
      aria-label="headArticle"
      class="head"
      style="background-image: url(./beanie.png); height: 150px; width: 150px; background-size: cover; background-repeat: no-repeat;"
    >
       
    </article>
    <article
      class="torso"
      style="background-image: url(./cardigan.png); height: 150px; width: 150px; background-size: cover; background-repeat: no-repeat;"
    >
       
    </article>
    <article
      class="legs"
      style="background-image: url(./baggy.png); height: 150px; width: 150px; background-size: cover; background-repeat: no-repeat;"
    >
       
    </article>
  </section>
</div>
`;
